starter-code/index.js

```diff
@@ -1,17 +1,55 @@
 class SortedList {
-  constructor() {}
+  constructor() {
+    this.items = [];
+    this.length = this.items.length;
+  }
 
-  add(item) {}
+  add(item) {
+    this.items.push(item);
+    this.length = this.items.length;
+    this.items.sort((a, b) => a - b);
+  }
 
-  get(pos) {}
+  get(pos) {
+    let item = this.items[pos];
+    if (pos > this.length) {
+      throw new Error('OutOfBounds');
+    }
+    return item;
+  }
 
-  max() {}
+  max() {
+    if (this.items == 0) {
+      throw new Error('EmptySortedList');
+    }
+    return this.items[this.length - 1];
+  }
 
-  min() {}
+  min() {
+    if (this.items == 0) {
+      throw new Error('EmptySortedList');
+    }
+    return this.items[0];
+  }
 
-  sum() {}
+  sum() {
+    let total = 0;
+    for (let i = 0; i < this.length; i++) {
+      total += this.items[i];
+    }
+    return total;
+  }
 
-  avg() {}
+  avg() {
+    let total = 0;
+    if (this.items == 0) {
+      throw new Error('EmptySortedList');
+    }
+    for (let i = 0; i < this.length; i++) {
+      total += this.items[i];
+    }
+    return total / this.length;
```

Comment:
You could have DRY(Don't repeat yourself)'ed this.

and just:

```js
return this.sum()/ this.items.length
```

Reply:
I tried this but somehow couldn't make it work! Something was missing and I didn't have the time to search. 
Thank you for all your comments :)